### PR TITLE
Update basic examples with namespaces

### PR DESCRIPTION
**Pull request by bistapf**
Updated the C++ analyzers code to the namespace format and adjusted basic example & documentation accordingly. Also included overview table + brief explanation of common functions used in readme.

### CODE REVIEW

**Comment by clementhelsens** [comment]
Thanks @bistapf , I ping @EmanuelPerez , @gganis , @patriziaazzi because I'll have very little time to look at this.

**Comment by patriziaazzi** [comment]
I cannot merge this pull request.

**Comment by clementhelsens** [comment]
I could hit the merge button for the time being no problem. But this PR needs to be reviewed before I think.

Le 22 mars 2021 à 09:45, Patrizia Azzi ***@***.******@***.***>> a écrit :



I cannot merge this pull request.

**Comment by patriziaazzi** [comment]
yes. I just sent the message bc I did not even had the possibility to merge.

Pat

Le 22 mars 2021 à 11:24, Clement Helsens ***@***.******@***.***>> a écrit :


I could hit the merge button for the time being no problem. But this PR needs to be reviewed before I think.

Le 22 mars 2021 à 09:45, Patrizia Azzi ***@***.******@***.***>> a écrit :



I cannot merge this pull request.

File: examples/basics/README.md [hunk 1]
```diff
@@ -44,17 +44,29 @@ The example read_EDM4HEP.py shows you how to access the different objects such a
...
 
-For the name of the object, in principle the names of the EDM4HEP collections are used - photons, muons and electrons are an exception where a few extra steps are required, as shown in the example here. 
+Below you can find an overview of the basic, most commonly required functions, to illustrate the naming conventions. This is not an exhaustive list, if you want to find out all functions that are available please take a look in the respective analyzers code itself - [here for reconstructed particles](https://github.com/bistapf/FCCAnalyses/blob/basicexamples/analyzers/dataframe/ReconstructedParticle.h) and [here for MC particles](https://github.com/bistapf/FCCAnalyses/blob/basicexamples/analyzers/dataframe/MCParticle.h).
```
Comment (gganis) [inline examples/basics/README.md:49]: Here there two links to a prvate repository (bistapf). Can these point to the files in the official directory? (If there are changes to those files, the changes should be perhaps parts of the PR?)

**Review by gganis — CHANGES_REQUESTED** [review]
Thanks, It looks good to me. Please fix the links to ReconstructedParticle.h and MCParticle.h (see specific comment).

File: examples/basics/README.md (hunk 1 above)
Comment (gganis) [reply examples/basics/README.md:49]: I can fix the links after the merge.

**Pull request merged**